06-building-a-program-start-to-finish/solving-the-problem/output/InvertedIndex.js

```diff
@@ -0,0 +1,233 @@
+          const response = InvertedIndex.validateFile(fileContent);
+          resolve(response);
+        };
+        reader.readAsText(file);
+      } catch (err) {
+        reject(err);
+      }
+    });
+  }
+
+  /**
+  *
+   *
+   * @param {Object} jsonContent - file content
+   * @returns{Object} isValid - returns true and JSON content for valid files
+   * and false for invalid files
+   * @memberOf InvertedIndex
+   */
+  static validateFile(jsonContent) {
+    let isValid = { status: false };
+    const invalidStructureError = {
+      name: 'validate file structure',
+      message: 'File structure is invalid'
+    };
+    if (Object.keys(jsonContent).length === 0 &&
+       typeof jsonContent === 'object') {
+      return isValid;
+    }
+    try {
+      jsonContent.forEach((doc) => {
+        if (!doc.title || !doc.text) {
+          throw invalidStructureError;
+        } else {
+          isValid = {
+            status: true,
+            jsonContent
+          };
+          return isValid;
+        }
+      });
+    } catch (error) {
+      if (error.name === 'validate file structure') return isValid;
+      throw error;
+    }
+    return isValid;
+  }
+  /**
+   *
+   *
+   * @static
+   * @param {Array} words -takes in an array of words with possible duplicates.
+   * @returns{Array} item -returns an array of words without duplicates.
+   *
+   * @memberOf InvertedIndex
+   */
+  static uniqueWords(words) {
+    if (Array.isArray(words)) {
+      const checked = {};
+      return words.filter((item) => {
+        if (!checked[item]) {
+          checked[item] = true;
+          return item;
+        }
+        return null;
+      });
+    }
+    return ['invalid data type supplied'];
+  }
+
+  /**
+   *
+   *
+   * @static
+   * @param {String} text - a string of spaced words with unwanted characters
+   * @returns{String} term -
+   *  a string of spaced words with no unwanted characters
+   *
+   * @memberOf InvertedIndex
+   */
+  static removeBadCharacters(text) {
+    const invalid = /[.,/#!$%^&*;:{}=\-_`~()]/g;
+    const term = text.replace(invalid, '');
+    return term;
+  }
+  /**
+   *
+   *
+   * @static
+   * @param {String} document
+   * @returns{Array} words - an array of unique words.
+   *
+   * @memberOf InvertedIndex
+   */
+  static tokenizeText(document) {
+    let words = document.toLowerCase().split(' ').sort();
+    words = InvertedIndex.uniqueWords(words);
+    return words;
+  }
+
+  /**
+   *
+   *
+   * @static
+   * @param {Object} document - single object of a JSON formatted file
+   * @returns{String} concatenatedText -
+   *  returns the concatenated values of object keys
+   *
+   * @memberOf InvertedIndex
+   */
+  static concatenateText(document) {
+    // if a document exists? combine title and text to split at once
+    let concatenatedText = '';
+    concatenatedText = `${document.title} ${document.text}`;
+    return concatenatedText;
+  }
```

Comment:
I believe this method `concatenateText()` is really not necessary. Consider moving line 143 to your `createIndex()`